core/arch/arm/mm/tee_mmu.c

```diff
@@ -648,6 +648,26 @@ TEE_Result vm_get_flags(struct user_mode_ctx *uctx, vaddr_t va, size_t len,
 	return TEE_SUCCESS;
 }
 
+TEE_Result vm_get_prot(struct user_mode_ctx *uctx, vaddr_t va, size_t len,
+		       uint16_t *attrs)
+{
+	struct vm_region *r = NULL;
+
+	if (!len || ((len | va) & SMALL_PAGE_MASK))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	r = find_vm_region(&uctx->vm_info, va);
+	if (!r)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (!va_range_is_contiguous(r, va, len, cmp_region_for_get_flags))
```

Comment:
`r->attr & TEE_MATTR_PROT_MASK` should be checked to be the same over the range rather than `f->flags`